ground/src/main/java/com/google/android/ground/ui/syncstatus/SyncStatusViewModel.kt

```diff
@@ -61,11 +62,16 @@ internal constructor(
 
   private suspend fun loadLocationsOfInterestAndPair(
     mutations: List<Mutation>
-  ): List<MutationDetail> = mutations.map { toMutationDetail(it) }
+  ): List<MutationDetail> = mutations.mapNotNull { toMutationDetail(it) }
 
-  private suspend fun toMutationDetail(mutation: Mutation): MutationDetail {
+  private suspend fun toMutationDetail(mutation: Mutation): MutationDetail? {
     val loi =
       locationOfInterestRepository.getOfflineLoi(mutation.surveyId, mutation.locationOfInterestId)
+    if (loi == null) {
```

Comment:
Any idea why this might be happening? Was the LOI removed remotely, or perhaps the ID in the local db changed?

Reply:
Nvm, I see the `LocationOfInterestMutation` references the LOI by UUID, not local DB primary key. 

If this happens, shouldn't we show the mutation as "failed" in the sync status screen? To do that, we may need to introduce a new type of error state in `SyncStatus`, `UNRECOVERABLE_ERROR` for ex. We could also rename `FAILED` to `PENDING_RETRY` or something clearer and update its current incorrect ktdoc in `SyncStatus`.